CONTRIBUTING.md

```diff
@@ -418,191 +351,1122 @@ This is called the `virtual path`. In this case it should be `stocks.dps.shorted
+
+    Why? Consistency and good code structure. This also improves the api user experience. Thus follows that view and model files will have the same arguments (except for output options like raw, export, external_axes), since no data changes shall be done in the view file.
+
+<br>
+
+2. Each model (get_) should almost always have its own view function (display_)
+
+    Why? To respect the principles laid out in Code Structure and the previous bullet point. If your code does not have this `get_` → `display_` map it’s likely that i. and/or ii. fail to hold.
+
+    i. Data is processed in _model files and displayed in `_view` files
+
+    ii. `_view` and `_model` files will have the same arguments (except for output options)
+
+<br>
+
+### Coding Style
+
+When in doubt, follow <https://www.python.org/dev/peps/pep-0008/>.
+
+#### OpenBB Style Guide
+
+The style guide is a reverse dictionary for argument names, where a brief definition is mapped to an OpenBB recommended argument name and type. When helpful a code snippet example is added below. Following this guide will help keep argument naming consistent and improve API users experience.
+
+Style guide structure:
+
+```python
+<definition> : <argument_name (argument_type)> e.g. <examples>
+
+def func(..., argument_name: argument_type = default, ...):
+    ...
+```
+
+<br>
+
+#### Flags
+
+Show raw data : `raw` *(bool)*
+
+```python
+def display_data(..., raw: bool = False, ...):
+    ...
+    if raw:
+        print_rich_table(...)
+```
+
+Sort in ascending order : `ascend` *(bool)*
+
+```python
+def display_data(..., sortby: str = "", ascend: bool = False, ...):
+    ...
+    if sortby:
+        data = data.sort_values(by=sortby, ascend=ascend)
+```
+
+Show plot : `plot` *(bool)*
+
+```python
+def display_data(..., plot: bool = False, ...):
+    ...
+    if plot:
+        ...
+        ax.plot(...)
+```
+
+<br>
+
+#### Output format
+
+Format to export data : `export` *(str), e.g. csv, json, xlsx*
+
+```python
+def display_data(..., export: str = "", ...):
+    ...
+    export_data(export, os.path.dirname(os.path.abspath(__file__)), "func", data)
+```
+
+List of external axes to include in a plot : `external_axes` *(Optional[List[plt.Axes]])*
+
+```python
+def display_data(..., external_axes: Optional[List[plt.Axes]] = None, ...):
+    ...
+    if external_axes is None:
+        _, ax = plt.subplots(figsize=plot_autoscale(), dpi=PLOT_DPI)
+    else:
+        ax = external_axes[0]
+    ax.plot(...)
+```
+
+Field by which to sort : `sortby` *(str), e.g. "Volume"*
+
+```python
+def display_data(..., sortby: str = "col", ...):
+    ...
+    if sortby:
+        data = data.sort_values(by=sortby)
+```
+
+Maximum limit number of output items : `limit` *(int)*
+
+```python
+def display_data(..., limit = 10, ...):
+    ...
+    print_rich_table(
+        data[:limit],
+        ...
+    )
+```
+
+<br>
+
+#### Time-related
+
+Date from which data is fetched (YYYY-MM-DD) : `start_date` *(str), e.g. 2022-01-01*
```

Comment:
Mention here that we want to accept date time in string so that is easier to deal from user perspective. And then we convert to date time and check if the date is valid within the function

Reply:
Done